### PR TITLE
Change arg of getSubGroups to briefRepresentation

### LINKED ISSUE #25053: GET `/admin/realms/{realm}/groups` Endpoint Not Returning `subGroups`, `clientRoles`, `realmRoles` Despite `populateHierarchy` and `briefRepresentation` settings"
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/api

### Describe the bug

The GET `/admin/realms/{realm}/groups` endpoint is not returning `subGroups`, `clientRoles`, and `realmRoles` as part of the response, despite the documentation indicating these are optional fields. The migration documentation mentions `populateHierarchy` is supposed to be `true` by default, but it seems to have no effect even if we pass it explicitely to `true`. Similarly, `briefRepresentation`, which was previously functional, now appears to be ineffective. This raises the question of whether this is a bug or an intentional change in the API functionality.

### Version

23.0

### Expected behavior

The endpoint should return `subGroups`, `clientRoles`, and `realmRoles` as part of the response when appropriate query parameters are set, as per the existing documentation and the default setting of populateHierarchy (`"populateHierarchy": true`).

### Actual behavior

Despite setting `populateHierarchy`, the endpoint response does not include subGroups, clientRoles, and realmRoles.

### How to Reproduce?

1. Send a GET request to `/admin/realms/{realm}/groups` with `populateHierarchy` set to `true`.
2. (OPTIONAL) Attempt the same with `briefRepresentation` set to true and false.
3. Observe the missing `subGroups`, `clientRoles`, and `realmRoles` in the response.

### Anything else?

We have reviewed the Keycloak migration documentation for version 23.0.0, specifically regarding `populateHierarchy` and the new endpoint for `Group Admin Api`, available here: [Migrating to 23.0.0](https://www.keycloak.org/docs/latest/upgrading/index.html#migrating-to-23-0-0). There is a lack of clarity on whether the absence of these fields in the response is a bug or an intended change. Clarification on this matter would be highly appreciated.

Here is the documentation for version `23.0`:  

GET on `/admin/realms/{realm}/groups`
image

Group Representation
image

### LINKED ISSUE #25096: Meaning of briefRepresentation query parameter is inverted in GroupResource.getSubGroups
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/api

### Describe the bug

While debugging a regression in keycloak-config-cli I noticed, that the meaning of the `briefRepresentation` parameter in https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/GroupResource.java#L162 has changed internally to mean the opposite. 

This was introduced here: https://github.com/keycloak/keycloak/commit/69497382d8cc6c622e815184d9bc27cb0dc6a400#diff-d233e82db465ac2b1ad22b124ab0bae3714f02d58c108c29afbbb935653d3ceaR162
As you can see the query parameter `briefRepresentation`  is mapped to the method parameter `full` and handled with the "meaning" of full.

I think this was a mistake as the parameter name "briefRepresentation" denotes the opposite of "full".

The current behaviour breaks all applications that rely on getting the full group data if briefRepresentation is set to false, but currently clients must use `briefRepresentation=true` to get the full representation.

### Version

23.0.0

### Expected behavior

Meaning of briefRepresentation should be consistent accross all Keycloak admin endpoints.

### Actual behavior

Meaning of briefRepresentation is inverted for GroupResource.getSubGroups.

### How to Reproduce?

1) Create a group with a subgroup
2) Add attributes and / or role mappings to the subgroup
2) try to obtain roles or attributes for the subgroup from the GroupRepresentation fetched with `briefRepresentation=false`

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by transducer**
Parameter name briefRepresentation should be briefRepresentation, not full. This way callers will by default get the full representation unless true is passed as value for briefRepresentation.

Fixes #25096

Mistake introduced in 23.0.0 in commit https://github.com/keycloak/keycloak/commit/69497382d8cc6c622e815184d9bc27cb0dc6a400#diff-d233e82db465ac2b1ad22b124ab0bae3714f02d58c108c29afbbb935653d3ceaR162.

I could not find a place to add an integration test for testing the endpoint with an external caller. If such a test exists please point me to it and I will add it.

### CODE REVIEW

**Comment by transducer** [comment]
@mhajas yes please let me know where I can add it. I actually don't think the current commit fixes https://github.com/keycloak/keycloak/issues/25053, since GET {keycloak server}/realms/{realm}/groups/{group_id}/children seems to be a new endpoint, and the issue states the older GET {keycloak server}/realms/{realm}/groups/ endpoint fails.

**Comment by mhajas** [comment]
@transducer Yes you are right. this does not fix #25053. Please remove it from the commit message and PR description so it is not linked to this PR. Also, Keycloak now requires [Developer's Certificate of Origin](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#developers-certificate-of-origin) otherwise the CI won't let this PR in.

Regarding the test, the correct file seems to be: https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/group/GroupTest.java

We are interested in doing a call like this: https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/group/GroupTest.java#L427 specifically the method `getSubGroups` with `briefRepresentation` set to `true` and `false` and then check the level of details within the returned list of `GroupRepresentation`s. Add a new test case for that please because some of the test cases already there are huge.

**Comment by mposolda** [comment]
@mhajas @transducer 2 questions:

- Do we want this change documented in the upgrading guide? Something like `In Keycloak 23.0.0 release, we accidentally changed the meaning of parameter in briefRepresentation. This is fixed now, so the briefRepresentation parameter has same meaning like before Keycloak 23 release` or something like this?

- Do we want this backported to Keycloak 23 as well, so it is available in next Keycloak 23.0.X release? I suppose that yes. In this case, the appropriate place for documenting this change might be file https://github.com/keycloak/keycloak/blob/main/docs/documentation/upgrading/topics/keycloak/changes-23_0_2.adoc

**Comment by transducer** [comment]
@mposolda The endpoint GET /groups/<group_id>/children was introduced in 23.0.0, this is also mentioned in the [previous release notes](https://github.com/keycloak/keycloak/commit/69497382d8cc6c622e815184d9bc27cb0dc6a400#diff-430c07d2628f0b5fc42d0999559c52d55c1972e64d4c814415c4248e41a4e23cR134). So I can add it to the notes for this PR.

As mentioned, the regression that the meaning was changed for other endpoints too (mentioned in https://github.com/keycloak/keycloak/issues/25053), will not be fixed by this PR.

I will add a unit test and a release note later today.

**Comment by mposolda** [comment]
@transducer Thanks!

**Comment by transducer** [comment]
@mposolda I've added a unit test and documentation.

I unfortunately made a mistake where I rebased a previous commit which caused extra reviewers to be automatically added. I removed this commit but the added labels stay.

I don't know if the test works, because I don't have a working Java IDE and locally I get a build error when running `mvn clean install`, and the CI does not run.

**Comment by ahus1** [comment]
@transducer - for first-time contributors the CI doesn't run automatically but has to be approved by one of the maintainers. This is the case for every new push to the PR until the you've had your first PR merged. This is a precaution built into standard GitHub. 

I've just now approved the run, and CI is starting.

**Comment by transducer** [comment]
@mposolda I think test should work now. Since I could not run the project locally I am using the CI to check. Could you run it once more?

**Comment by transducer** [comment]
@mposolda sorry it's a bit inconvenient using the CI to check, the order in the check was wrong, now using hamcrest.Matchers it should be fine. Could it be started once again?

**Comment by transducer** [comment]
@ahus1 could you approve one more run of this PR? I think the test is okay now (cannot test locally).

**Comment by mposolda** [comment]
@transducer Sorry, one small thing: Keycloak 23.0.2 has already code-freeze for past days, so this PR won't be possible to catch in that release. Is it possible to put it to `changes-23_0_3.adoc` instead? Also might be needed to include this file from `changes.adoc` as it is not created yet.

**Comment by transducer** [comment]
@mposolda done

**Comment by transducer** [comment]
@mposolda Now Store Models Test is failing ([logs](https://github.com/keycloak/keycloak/actions/runs/7168641214/job/19587997734?pr=25264#logs)). I searched for "Errors: 1" (Failures are all 0) and found: `7:56:50,408 ERROR [org.keycloak.models.sessions.infinispan.initializer.InfinispanCacheInitializer] (pool-50-thread-3) Interruped exception when computed future. Errors: 1` I don't think this is related to my change?

**Review by mposolda — APPROVED** [review]
@transducer Yes, I also think it is not related to your change. Will try to re-run. Hope it helps...

**Pull request merged**
